lib/jobs/message_latency_report.ex

```diff
@@ -210,13 +198,27 @@ defmodule Jobs.MessageLatencyReport do
     id
   end
 
-  defp get_percentile_row(percentile_index, rows_with_index) do
-    {percentile_row, _index} =
-      Enum.find(rows_with_index, {[], nil}, fn {_row, index} ->
-        index == percentile_index
-      end)
+  defp get_percentiles(rows_with_index) do
+    num_rows = Enum.count(rows_with_index)
+    percentile_95_index = calculate_percentile_index(num_rows, @percentile_95)
+    percentile_99_index = calculate_percentile_index(num_rows, @percentile_99)
 
-    percentile_row
+    {Enum.reduce_while(rows_with_index, {[], []}, fn {current_row, index},
+                                                     {percentile_95, percentile_99} ->
+       cond do
+         percentile_95 != [] and percentile_99 != [] ->
+           {:halt, {percentile_95, percentile_99}}
+
+         index == percentile_95_index ->
+           {:cont, {current_row, percentile_99}}
+
+         index == percentile_99_index ->
```

Comment:
This may never be an issue, but if the data set is small enough, the 95th and 99th percentile could be the same, in which case the `cond` would end up skipping the 99th. If we wanted to fix that we'd need to e.g. pipe the accumulator through both checks.

Reply:
Hmm yeah good point. It's certainly possible considering the amount of data that ARINC as sent us varies wildly from past experience. I'll make a change to make sure we don't skip the 99th